sov-modules/sov-state/src/jmt_storage.rs

```diff
@@ -67,9 +73,14 @@ impl JmtStorage {
 
         self.cache.borrow().get_first_reads()
     }
+
+    /// Take the log
+    pub fn take_treedb_log(&self) -> Option<TreeReadLogger> {
+        self.read_logger.borrow_mut().take()
+    }
 }
 
-impl Storage for JmtStorage {
+impl<H: SimpleHasher> Storage for JmtStorage<H> {
```

Comment:
What about parameterizing only the `finalize` fn by SimpleHasher instead of the whole struct?  This way we don't have to add generic parameter everywhere. (Not for this draft, but maybe we could consider it when we bring this change over to `research`)